src/main/java/io/vertx/core/net/impl/SSLHelper.java

```diff
@@ -321,27 +321,51 @@ private TrustManagerFactory getTrustMgrFactory(VertxInternal vertx) throws Excep
       return null;
     }
     if (crlPaths != null && crlValues != null && (crlPaths.size() > 0 || crlValues.size() > 0)) {
-      Stream<Buffer> tmp = crlPaths.
-          stream().
-          map(path -> vertx.resolveFile(path).getAbsolutePath()).
-          map(vertx.fileSystem()::readFileBlocking);
+      Stream<Buffer> tmp = readCrlFromFile(vertx);
       tmp = Stream.concat(tmp, crlValues.stream());
-      CertificateFactory certificatefactory = CertificateFactory.getInstance("X.509");
-      ArrayList<CRL> crls = new ArrayList<>();
-      for (Buffer crlValue : tmp.collect(Collectors.toList())) {
-        crls.addAll(certificatefactory.generateCRLs(new ByteArrayInputStream(crlValue.getBytes())));
-      }
-      TrustManager[] mgrs = createUntrustRevokedCertTrustManager(fact.getTrustManagers(), crls);
+      crls = generateCrl(tmp);
+      TrustManager[] mgrs = createUntrustRevokedCertTrustManager(fact.getTrustManagers());
       fact = new VertxTrustManagerFactory(mgrs);
     }
     return fact;
   }
 
+  public void reloadCrlFromPath(final VertxInternal vertx) throws CertificateException, CRLException {
+    if (crlPaths == null || crlPaths.size() == 0) {
+      throw new VertxException("No CRL path were provided in initial configuration. Won't reload any CRL file.");
+    }
+    Stream<Buffer> stream = readCrlFromFile(vertx);
+    crls = generateCrl(stream);
+  }
+
+  public void reloadCrlFromBuffer(final List<Buffer> buffers) throws CertificateException, CRLException {
+    if (buffers == null || buffers.size() == 0) {
+      throw new VertxException("Can't reload CRL from empty buffer.");
+    }
+    crls = generateCrl(buffers.stream());
+  }
+
+  private List<CRL> generateCrl(final Stream<Buffer> stream) throws CertificateException, CRLException {
+    List<CRL> crls = new ArrayList<>();
+    CertificateFactory certificatefactory = CertificateFactory.getInstance("X.509");
+    for (Buffer crlValue : stream.collect(Collectors.toList())) {
+      crls.addAll(certificatefactory.generateCRLs(new ByteArrayInputStream(crlValue.getBytes())));
+    }
+    return crls;
+  }
+
+  private Stream<Buffer> readCrlFromFile(final VertxInternal vertx) {
```

Comment:
this should be non blocking as the reload might be called from the event loop initially